app/src/common/shared/org/mozilla/vrbrowser/telemetry/TelemetryWrapper.java

```diff
@@ -292,5 +344,217 @@ public static void uploadImmersiveToHistogram() {
+        TelemetryEvent event = TelemetryEvent.create(Category.ACTION, Method.CURVED_MODE_ACTIVE, Object.WINDOW);
+        event.queue();
+
+        Log.d(LOGTAG, "[Queue] Curved mode active");
+    }
+
+    private static void queueWindowsLifetimeHistogram() {
+        Log.d(LOGTAG, "[Queue] Windows Lifetime Histogram:");
+        queueHistogram(windowsLifetimeHistogram, Method.WINDOW_LIFETIME, Object.WINDOW);
+        windowsLifetimeHistogram = new TelemetryHistogram(HISTOGRAM_SIZE, MULTI_WINDOW_BIN_SIZE_MS, 0);
+    }
+
+    private static void queueWindowsMovesCountEvent() {
+        float movesFreqPerSession = (float)windowsMovesCount / ((SystemClock.elapsedRealtime() - sessionStartTime)/1000);
+        TelemetryEvent event = TelemetryEvent.create(Category.ACTION, Method.WINDOWS_MOVES_FREQ, Object.WINDOW);
+        event.extra(Extra.WINDOW_MOVES_FREQ, String.valueOf(movesFreqPerSession));
+        event.queue();
+
+        Log.d(LOGTAG, "[Queue] Windows Moves Freq: " + movesFreqPerSession);
+
+        windowsMovesCount = 0;
+    }
+
+    private static void queueWindowsResizesCountEvent() {
+        float resizesFreqPerSession = (float)windowsResizesCount / ((SystemClock.elapsedRealtime() - sessionStartTime)/1000);
+        TelemetryEvent event = TelemetryEvent.create(Category.ACTION, Method.WINDOWS_RESIZE_FREQ, Object.WINDOW);
+        event.extra(Extra.WINDOW_RESIZE_FREQ, String.valueOf(resizesFreqPerSession));
+        event.queue();
+
+        Log.d(LOGTAG, "[Queue] Windows Resizes Freq: " + resizesFreqPerSession);
+
+        windowsResizesCount = 0;
+    }
+
+    private static void queueActiveWindowPctEvent() {
+        long totalTime = 0;
+        for (long time : activePlacementTime)
+            totalTime += time;
+
+        if (totalTime == 0)
+            return;
+
+        float[] pcts = new float[MAX_WINDOWS];
+        for (int index = 0; index< MAX_WINDOWS; index++)
+            pcts[index] = (activePlacementTime[index]*100.0f)/totalTime;
+
+        TelemetryEvent event = TelemetryEvent.create(Category.ACTION, Method.PLACEMENTS_ACTIVE_PCT, Object.WINDOW);
+        event.extra(Extra.LEFT_WINDOW_ACTIVE_PCT, String.valueOf(pcts[WindowPlacement.LEFT.getValue()]));
+        event.extra(Extra.FRONT_WINDOW_ACTIVE_PCT, String.valueOf(pcts[WindowPlacement.FRONT.getValue()]));
+        event.extra(Extra.RIGHT_WINDOW_ACTIVE_PCT, String.valueOf(pcts[WindowPlacement.RIGHT.getValue()]));
+        event.queue();
+
+        Log.d(LOGTAG, "[Queue] Placements Active Pct:");
+        Log.d(LOGTAG, "\tFRONT: " + pcts[WindowPlacement.FRONT.getValue()]);
+        Log.d(LOGTAG, "\tLEFT: " + pcts[WindowPlacement.LEFT.getValue()]);
+        Log.d(LOGTAG, "\tRIGHT: " + pcts[WindowPlacement.RIGHT.getValue()]);
+
+        for (int index = 0; index< MAX_WINDOWS; index++) {
+            activePlacementTime[index] = 0;
+            activePlacementStartTime[index] = 0;
+        }
+    }
+
+    private static void queueOpenWindowsAvgEvent() {
```

Comment:
Should we measure time here too (or in a extra data) as you are doing in `queueActiveWindowPctEvent`? You can have 1.5 windows open in a session but how do you know how long they have used 1 windows or 2 windows?

Reply:
I think the goal of that stat is knowing what's the average of open windows in a session but we can also make a a % of how much time the user spends with 1, 2 or 3 open windows. It just depends of what's more useful for us.

Reply:
% of how much time the user spends with 1, 2 or 3 open windows would be great! probably more useful than the average number.

Reply:
@jvonitter ok, I'll add both :)